src/encoding.rs

```diff
@@ -1,20 +1,17 @@
-use std::convert::{TryInto};
-
-pub type Padding = u32;
-pub const PADDING: usize = std::mem::size_of::<Padding>();
+use integer_encoding::VarInt;
 
 /// Encode a message, returning the bytes that must be sent before the message.
-pub fn encode_size(message: &[u8]) -> [u8; PADDING] {
-    (message.len() as Padding).to_le_bytes()
+/// A buffer is used to avoid heap allocation.
+/// It is size 10=ceil(64/7), the max required bytes to encode a u64.
+pub fn encode_size<'a>(message: &[u8], buf: &'a mut [u8; 10]) -> &'a [u8] {
+    let varint_size = message.len().encode_var(buf);
```

Comment:
`(message.len() as u64).encode_var(buf)` to ensure compatibility with x32 machines